### PR TITLE
Update Performance Max examples for listing groups and signals

### PR DESCRIPTION
**Pull request by jradcliff**
* Add listing group filters in Performance Max retail campaign example
* Add asset group signals in Performance Max campaign example

### CODE REVIEW

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/shoppingads/AddPerformanceMaxRetailCampaign.java [hunk 1]
```diff
@@ -124,8 +128,11 @@
 
     @Parameter(
         names = ArgumentNames.FINAL_URL,
-        description = "The final URL for the asset group of the campaign.")
-    private String finalUrl = "http://www.example.com";
+        required = true,
+        description =
+            "The final url for the generated ads. Must have the same domain as the Merchant Center"
+                + " account.")
+    private String finalUrl;
```
Comment (PierrickVoulet) [inline google-ads-examples/src/main/java/com/google/ads/googleads/examples/shoppingads/AddPerformanceMaxRetailCampaign.java:135]: Making this parameter a mandatory one makes sense to me but it does not seem to be in the current version of C# (golden) and PHP. Should we communicate this to all?

**Review by PierrickVoulet — CHANGES_REQUESTED** [review]

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/shoppingads/AddPerformanceMaxRetailCampaign.java (hunk 1 above)
Comment (PierrickVoulet) [reply google-ads-examples/src/main/java/com/google/ads/googleads/examples/shoppingads/AddPerformanceMaxRetailCampaign.java:135]: You are correct. Looks like we missed that for PHP though, I will fix.

Comment (jradcliff) [reply google-ads-examples/src/main/java/com/google/ads/googleads/examples/shoppingads/AddPerformanceMaxRetailCampaign.java:135]: Thanks!

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/shoppingads/AddPerformanceMaxRetailCampaign.java [hunk 2]
```diff
@@ -658,6 +687,39 @@ private MutateOperation createPerformanceMaxCampaignOperation(
...
+    AssetGroupListingGroupFilter listingGroupFilter =
+        AssetGroupListingGroupFilter.newBuilder()
+            .setAssetGroup(assetGroupResourceName)
+            // Does not set the parentListingGroupFilter since this is the root node. For all other
+            // nodes, this would refer to the parent listing group filter resource name.
+            // .setParentListingGroupFilter("<PARENT FILTER RESOURCE NAME>")
+
+            // Sets the type to UNIT_INCLUDED since this node has no children.
+            .setType(ListingGroupFilterType.UNIT_INCLUDED)
+            // Specifies that this is in the shopping vertical, as required for a Performance Max
```
Comment (devchas) [inline google-ads-examples/src/main/java/com/google/ads/googleads/examples/shoppingads/AddPerformanceMaxRetailCampaign.java:707]: Nit: SHOPPING

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/shoppingads/AddPerformanceMaxRetailCampaign.java [hunk 3]
```diff
@@ -658,6 +687,39 @@ private MutateOperation createPerformanceMaxCampaignOperation(
...
+            .build();
+
+    // Adds an operation to the list to create the listing group filter.
+    mutateOperations.add(
+        MutateOperation.newBuilder()
+            .setAssetGroupListingGroupFilterOperation(
+                AssetGroupListingGroupFilterOperation.newBuilder().setCreate(listingGroupFilter))
+            .build());
+
+    return mutateOperations;
```
Comment (devchas) [inline google-ads-examples/src/main/java/com/google/ads/googleads/examples/shoppingads/AddPerformanceMaxRetailCampaign.java:719]: We can shorten this to just return the single mutate operation as a list and remove line 694. It seems unnecessary to create a list, add to it, then return the list since there's only one item in the list.

**Review by devchas — APPROVED** [review]
